DWIConvert/DWIConvertLib.cxx

```diff
@@ -279,18 +279,20 @@ DWIConverter * DWIConvert::CreateDicomConverter(
   return converter;
 }
 
-//one of ["DicomToNrrd", "DicomToFSL", "NrrdToFSL", "FSLToNrrd", "NrrdToNrrd", "FSLToFSL"]
-//if return "", invalidate input parameters.
 
+//currently supported file types: { ".nii", ".nii.gz", ".nhdr", ".nrrd"}
 void DWIConvert::setInputFileType(const std::string& inputVolume, const std::string& inputDicomDirectory){
   m_inputVolume = inputVolume;
   m_inputDicomDirectory = inputDicomDirectory;
-  if ("" == m_inputDicomDirectory){
-    std::string inputExt = findFilenameExt(m_inputVolume);
-    if (".nii" == inputExt) m_inputFileType = "FSL";
-    if (".nrrd" == inputExt || ".nhdr" == inputExt) m_inputFileType = "Nrrd";
+  if ("" == m_inputDicomDirectory && "" != m_inputVolume){
+    std::string inputExt = itksys::SystemTools::GetFilenameExtension(m_inputVolume);
+    if ( std::string::npos != inputExt.rfind(".nii")) m_inputFileType = "FSL";
+    else if (std::string::npos != inputExt.rfind(".nrrd") || std::string::npos != inputExt.rfind(".nhdr")) m_inputFileType = "Nrrd";
+    else {
+      std::cerr <<"Error: file type of inputVoume is not supported currently"<<std::endl;
+    }
   }
-  else if ("" == m_inputVolume)
+  else if ("" != m_inputDicomDirectory)
```

Comment:
add 'const std::string emptyString(""); //A named empty string.' somewhere up above, and then use that constant variable thoughout.